spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/microsoft/teams/MicrosoftTeamsNotifierTest.java

```diff
@@ -0,0 +1,231 @@
+        assertEquals("Summary doesn't match", MESSAGE_SUMMARY, testMessage.getSummary());
+
+        Section expectedSection = new Section();
+        expectedSection.setActivityTitle(stubApp.getName());
+        expectedSection.setActivityImage(testMTNotifier.getImageUrl().toString());
+        expectedSection.setActivitySubtitle(String.format(DEREGISTER_ACTIVITY_SUBTITLE_PATTERN,
+                stubApp.getName(),
+                stubApp.getId()));
+        expectedSection.getFacts().add(new Fact(STATUS_KEY, stubApp.getStatusInfo().getStatus()));
+        expectedSection.getFacts().add(new Fact(SERVICE_URL_KEY, stubApp.getServiceUrl()));
+        expectedSection.getFacts().add(new Fact(HEALTH_URL_KEY, stubApp.getHealthUrl()));
+        expectedSection.getFacts().add(new Fact(MANAGEMENT_URL_KEY, stubApp.getManagementUrl()));
+        expectedSection.getFacts().add(new Fact(SOURCE_KEY, stubApp.getSource()));
+
+
+        assertEquals("Incorrect number of sections", 1, testMessage.getSections().size());
+        assertEquals("Sections don't match", expectedSection, testMessage.getSections().get(0));
+    }
+
+    @Test
+    public void test_getRegisteredMessageForAppReturns_correctContent() {
+        Message testMessage = testMTNotifier.getRegisteredMessage(stubApp);
+
+        assertEquals("Title doesn't match", REGISTERED_TITLE, testMessage.getTitle());
+        assertEquals("Summary doesn't match", MESSAGE_SUMMARY, testMessage.getSummary());
+
+        Section expectedSection = new Section();
+        expectedSection.setActivityTitle(stubApp.getName());
+        expectedSection.setActivityImage(testMTNotifier.getImageUrl().toString());
+        expectedSection.setActivitySubtitle(String.format(REGISTER_ACTIVITY_SUBTITLE_PATTERN,
+                stubApp.getName(),
+                stubApp.getId()));
+        expectedSection.getFacts().add(new Fact(STATUS_KEY, stubApp.getStatusInfo().getStatus()));
+        expectedSection.getFacts().add(new Fact(SERVICE_URL_KEY, stubApp.getServiceUrl()));
+        expectedSection.getFacts().add(new Fact(HEALTH_URL_KEY, stubApp.getHealthUrl()));
+        expectedSection.getFacts().add(new Fact(MANAGEMENT_URL_KEY, stubApp.getManagementUrl()));
+        expectedSection.getFacts().add(new Fact(SOURCE_KEY, stubApp.getSource()));
+
+        assertEquals("Incorrect number of sections", 1, testMessage.getSections().size());
+        assertEquals("Sections don't match", expectedSection, testMessage.getSections().get(0));
+    }
+
+    @Test
+    public void test_getStatusChangedMessageForAppReturns_correctContent() {
+        Message testMessage = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofUp(), StatusInfo.ofDown());
+
+        assertEquals("Title doesn't match", STATUS_CHANGED_TITLE, testMessage.getTitle());
+        assertEquals("Summary doesn't match", MESSAGE_SUMMARY, testMessage.getSummary());
+
+        Section expectedSection = new Section();
```

Comment:
Instead of creating a expected section and comparing the actual against it, please assert the attributes of the actual section against their expected values. This way you see direct which attribute differs instead of just knowing that some of the many attributes is wrong! (and also you don't need to implement and test equals/hashcode)